File: 백준/Bronze/2490. 윷놀이/윷놀이.cc
#include <bits/stdc++.h>

using namespace std;

int main() {
    
    ios::sync_with_stdio(0);
    cin.tie(0);

    int numCnt = 4;
    int playCnt = 0;
    vector<int> numbers;

    // 3회 반복
    while(playCnt < 3)
    {

        
        // 윳 숫자 입력받기
        for(int i = 0; i < numCnt; i++)
        {
            int num = 0;
            cin >> num;
            numbers.push_back(num);
        }


        // 윳 숫자 정렬
        sort(numbers.begin(), numbers.end());

        bool isDo = numbers[0] == 0 && numbers[0] != numbers[1];
        bool isGae = numbers[0] == 0 && numbers[1] != numbers[2];
        bool isGurl = numbers[0] == 0 && numbers[2] != numbers[3];
        bool isYut = numbers[0] == 0 && numbers[3] == 0;
        bool isMo = numbers[0] == 1 && numbers[3] == 1;
    
        if(isDo) cout << "A" << '\n';
        else if(isGae) cout << "B"<< '\n';
        else if(isGurl) cout << "C"<< '\n';
        else if(isYut) cout << "D"<< '\n';
        else if(isMo) cout << "E"<< '\n';
        
        numbers.clear();
        playCnt++;
    }
}

File: 백준/Bronze/2490. 윷놀이/윷놀이.cs
using System;
using System.Linq;

namespace MyCompiler {
    class Program {
        public static void Main(string[] args) {

            int playCnt = 0;
            
            while(playCnt < 3)
            {
                // 윳 값 입력받기
                var input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

                // 윳 값 정렬
                var sortValues = input.OrderBy(value => value).ToArray();

                // 윷 값으로 반환되는 값 확인하는 조건
                bool isDo = sortValues[0] == 0 && sortValues[0] != sortValues[1];
                bool isGae = sortValues[1] == 0 && sortValues[1] != sortValues[2];
                bool isGurl = sortValues[2] == 0 && sortValues[2] != sortValues[3];
                bool isYut = sortValues[0] == 0 && sortValues[0] == sortValues[3];
                bool isMo = sortValues[0] == 1 && sortValues[0] == sortValues[3];

                if(isDo) Console.WriteLine("A");
                else if(isGae) Console.WriteLine("B");
                else if(isGurl) Console.WriteLine("C");
                else if(isYut) Console.WriteLine("D");
                else if(isMo) Console.WriteLine("E");

                // 플레이 횟수 증가
                playCnt++;
            }
        }
    }
}